From adffb16fc628859f98e32ebfd01f1c67920bc5c7 Mon Sep 17 00:00:00 2001
From: David Bauer <mail@david-bauer.net>
Date: Thu, 13 Feb 2025 17:00:44 +0100
Subject: [PATCH 6/8] wds: implement selection of default profile

---
 uqmi/commands-wds.c | 23 +++++++++++++++++++++++
 uqmi/commands-wds.h |  2 ++
 2 files changed, 25 insertions(+)

diff --git a/uqmi/commands-wds.c b/uqmi/commands-wds.c
index f57f126..6657a42 100644
--- a/uqmi/commands-wds.c
+++ b/uqmi/commands-wds.c
@@ -649,6 +649,29 @@ cmd_wds_get_default_profile_cb(struct qmi_dev *qmi, struct qmi_request *req, str
 	blobmsg_close_table(&status, p);
 }
 
+#define cmd_wds_set_default_profile_cb no_cb
+
+static enum qmi_cmd_result
+cmd_wds_set_default_profile_prepare(struct qmi_dev *qmi, struct qmi_request *req, struct qmi_msg *msg, char *arg) {
+	struct uqmi_wds_profile_identifier profile;
+
+	if (uqmi_wds_profile_identifier_parse(arg, &profile) < 0) {
+		fprintf(stderr, "Invalid argument\n");
+		return QMI_CMD_EXIT;
+	}
+
+	struct qmi_wds_set_default_profile_number_request set_default_profile = {
+		QMI_INIT_SEQUENCE(profile_identifier,
+			.family = QMI_WDS_PROFILE_FAMILY_TETHERED,
+			.type = profile.type,
+			.index = profile.index,
+		)
+	};
+
+	qmi_set_wds_set_default_profile_number_request(msg, &set_default_profile);
+	return QMI_CMD_REQUEST;
+}
+
 #define cmd_wds_delete_profile_cb no_cb
 
 static enum qmi_cmd_result
diff --git a/uqmi/commands-wds.h b/uqmi/commands-wds.h
index 043cc64..ecc36a7 100644
--- a/uqmi/commands-wds.h
+++ b/uqmi/commands-wds.h
@@ -34,6 +34,7 @@
 	__uqmi_command(wds_set_autoconnect_settings, set-autoconnect, required, QMI_SERVICE_WDS), \
 	__uqmi_command(wds_reset, reset-wds, no, QMI_SERVICE_WDS), \
 	__uqmi_command(wds_get_profile_settings, get-profile-settings, required, QMI_SERVICE_WDS), \
+	__uqmi_command(wds_set_default_profile, set-default-profile, required, QMI_SERVICE_WDS), \
 	__uqmi_command(wds_get_default_profile, get-default-profile, required, QMI_SERVICE_WDS), \
 	__uqmi_command(wds_create_profile, create-profile, required, QMI_SERVICE_WDS), \
 	__uqmi_command(wds_modify_profile, modify-profile, required, QMI_SERVICE_WDS), \
@@ -61,6 +62,7 @@
 		"  --set-ip-family <val>:            Set ip-family (ipv4, ipv6, unspecified)\n" \
 		"  --set-autoconnect <val>:          Set automatic connect/reconnect (disabled, enabled, paused)\n" \
 		"  --get-profile-settings <val,#>:   Get APN profile settings (3gpp, 3gpp2),#\n" \
+		"  --set-default-profile <val,#>:    Set default profile number (3gpp, 3gpp2)\n" \
 		"  --get-default-profile <val>:      Get default profile number (3gpp, 3gpp2)\n" \
 		"  --create-profile <val>            Create profile (3gpp, 3gpp2)\n" \
 		"    --apn <apn>:                    Use APN\n" \
-- 
2.47.2

